doc/internal/man7/removal.pod

```diff
@@ -0,0 +1,80 @@
+=pod
+
+=head1 NAME
+
+removal - How to do Public API removal
+
+=head1 DESCRIPTION
+
+Removal of a public function should only be done after it has been
+previously deprecated as documented in L<deprecation(7)>.
+
+Once you are ready to remove a deprecated function, you need to
+ensure one or more of the following things is true:
+
+=over 4
+
+=item There is a usually straightforward one-to-one replacement of the
+deleted function with a new function.
+
+=item The existing documentation for the function in the last release
+describes how to replace the function. If you feel this documentation
+is inadequate or incomplete, consider updating it if a maintenance
+release will happen.
+
+=item Details on how to replace the function are provided in
+L<ossl-guide-migration(7)>. If the replacement is complex and was
+not adequately documented in the past release, it is a good idea
+to write a reasonable section here.
+
+=back
+
+Once that is done, you should prepare your PR that removes the
+function, internal uses, and documentation.
+
+Your PR removing functions should include adding the function to the
+list in L<ossl-removed-api(7)> with an appropriate entry. Keep it short and
+simple. Each entry in the list for a version can include a quick link
+to the usual replacement such as:
+
+=over 4
```

Comment:
It would be useful to keep a "changelog style" entry listing functions by version they were removed in; example:

```
Removed in OpenSSL 4.0
- foo (Deprecated in 3.0) see foo-replacement{3)
- goat (Deprecated in 1.1.1) see goat-replacement{3)
```